### PR TITLE
refactor(zk.js): add program utxo type

### PR DESCRIPTION
**Pull request by ananas-block**
### Issue:
- this is the second utxo type refactor pr

### Changes:
- added ProgramOutUtxo type and helper functions plus tests
   - ProgramOutUtxo
   - createProgramOutUtxo
   - checkUtxoData (throws an error if a utxo data field is missing)
   - createUtxoDataHash
   - programOutUtxoToBytes
   - programOutUtxoFromBytes
   - programOutUtxoFromString
   - programOutUtxoToString
   - encryptProgramOutUtxo
   - decryptProgramOutUtxo
   - ProgramUtxo
   - createProgramUtxo
   - decryptProgramUtxo
   - programOutUtxoToProgramUtxo

### CODE REVIEW

File: zk.js/tests/testData/tmp_test_psp -legacy.ts [hunk 1]
```diff
@@ -0,0 +1,1142 @@
+export type TmpTestPsp = {
```
Comment (vadorovsky) [inline zk.js/tests/testData/tmp_test_psp -legacy.ts:1]: can we rename the file to `tmp_test_psp_legacy.ts` or something similar without spaces? Typing this name in terminal is gonna be painful (`tmp_test_psp\ -legacy.ts` and on some terminals even that doesn't work)

File: zk.js/tests/utxo.test.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import {
 } from "@solana/web3.js";
 import { BN } from "@coral-xyz/anchor";
 import { it } from "mocha";
-import { IDL as TEST_PSP_IDL } from "./testData/tmp_test_psp";
+import { IDL as TEST_PSP_IDL } from "./testData/tmp_test_psp -legacy";
```
Comment (vadorovsky) [inline zk.js/tests/utxo.test.ts:9]: :smiling_face_with_tear:

File: zk.js/tests/testData/tmp_test_psp.ts [hunk 3]
```diff
@@ -333,7 +333,7 @@ export type TmpTestPsp = {
       };
     },
     {
-      name: "utxo";
+      name: "utxoOutUtxo";
```
Comment (SwenSchaeferjohann) [inline zk.js/tests/testData/tmp_test_psp.ts:336]: why utxo twice in the name?

File: zk.js/src/programUtxo.ts [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+  hasher: Hasher,
+  compressed: boolean = true,
+  merkleProof: string[],
+  merkleTreeLeafIndex: number,
+  assetLookupTable: string[],
+  pspId: PublicKey,
+  pspIdl: Idl,
+  utxoName: string,
+): Promise<Result<ProgramUtxo | null, UtxoError>> {
+  const decryptedProgramOutUtxo = await decryptProgramOutUtxo({
```
Comment (SwenSchaeferjohann) [inline zk.js/src/programUtxo.ts:442]: given the large amount of parameters i'd suggest we use object destructuring syntax

Comment (SwenSchaeferjohann) [reply zk.js/src/programUtxo.ts:442]: i.e. ({ encBytes: ... , account: ..} : { encBytes: Uint8Array; account: Account,...})

File: zk.js/tests/testData/tmp_test_psp.ts (hunk 3 above)
Comment (ananas-block) [reply zk.js/tests/testData/tmp_test_psp.ts:336]: Because the idl generates a type and an object with the same fields.

**Review by SwenSchaeferjohann — COMMENTED** [review]
looks fine to me beyond the comments and michals comments

**Review by vadorovsky — APPROVED** [review]

**Pull request merged**